.github/workflows/build_and_test_snap.yaml

```diff
@@ -0,0 +1,67 @@
+# Note: The trigger for GPU tests is based off the name of this workflow.
+#       If we rename this workflow, update the GPU test trigger as well.
+name: Build and test Snap
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+env:
+  SNAP_ARTIFACT_NAME: dss-snap
+  SNAP_FILE: dss.snap
+  KUBECONFIG: ~/.dss/config
+
+jobs:
+  build-snap:
+    name: Build the snap
+    runs-on: ubuntu-22.04
+    outputs:
+      snap-file: ${{ steps.snapcraft.outputs.snap }}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - uses: snapcore/action-build@v1.2.0
+      id: snapcraft
+      with:
+        # Use a deterministic file name, so we can easily reference it later
+        snapcraft-args: "-o ${{ env.SNAP_FILE }}"
+
+    - uses: actions/upload-artifact@v4
+      id: upload-snap-artifact
+      with:
+        name: ${{ env.SNAP_ARTIFACT_NAME }}
+        path: ${{ steps.snapcraft.outputs.snap }}
+
+  test:
+    name: Test the snap
+    runs-on: ubuntu-22.04
+    needs:
+      - build-snap
+    steps:
+    - uses: actions/download-artifact@v4
+      id: download-snap-artifact
+      with:
+        name: ${{ env.SNAP_ARTIFACT_NAME }}
+
+    - name: Install the snap locally
+      # use --dangerous so we trust the snap even though it doesn't have signatures
+      run: |
+        sudo snap install ${{ env.SNAP_FILE }} --dangerous
+        sudo snap connect data-science-stack:dot-dss-config
+
+    - name: Set up Microk8s
+      uses: balchua/microk8s-actions@v0.3.2
+      with:
+        channel: '1.28/stable'
+        addons: '["hostpath-storage"]'
+
+    - name: Export kubeconfig where strict snap can access it
+      run: |
+        mkdir -p ~/.dss
+        cp ~/.kube/config ${{ env.KUBECONFIG }}
+
+    - name: Test dss snap
+      run: |
+        data-science-stack.dss initialize --kubeconfig ${{ env.KUBECONFIG }}
+        data-science-stack.dss create x --image kubeflownotebookswg/jupyter-scipy:v1.8.0 --kubeconfig ${{ env.KUBECONFIG }}
```

Comment:
Is there a reason why we dont create the alias ? 
```
sudo snap alias  data-science-stack.dss dss
```

Reply:
Omitted the alias before since it doesn't really change what is being tested, but now that we've moved the tests to a tox environment that directly calls `dss` as the executable I've added the alias in above in the snap building step